genesyscloud/group_roles/resource_genesyscloud_group_roles.go

```diff
@@ -0,0 +1,71 @@
+package group_roles
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v119/platformclientv2"
+)
+
+/*
+The resource_genesyscloud_group_roles.go contains all of the methods that perform the core logic for a resource
+*/
+
+func createGroupRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	groupID := d.Get("group_id").(string)
+	d.SetId(groupID)
+	return updateGroupRoles(ctx, d, meta)
+}
+
+// readGroupRoles is used by the group_roles resource to read Group Roles from the genesys cloud
+func readGroupRoles(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*genesyscloud.ProviderMeta).ClientConfig
+	authAPI := platformclientv2.NewAuthorizationApiWithConfig(sdkConfig)
```

Comment:
We need to add a Proxy class.  We should not be calling the API directly from the code.